lib/open_api_spex/plug/swagger_ui.ex

```diff
@@ -117,8 +117,37 @@ defmodule OpenApiSpex.Plug.SwaggerUI do
           request.headers["x-csrf-token"] = "<%= csrf_token %>";
           return request;
         },
-        displayOperationId: <%= display_operation_id %>
-      })
+        displayOperationId: <%= display_operation_id %>,
+        <%= if oauth[:redirect_url] do %>
+          oauth2RedirectUrl: "<%= oauth[:redirect_url] %>"
+        <% end %>
+      });
+      <%= if oauth do %>
+        ui.initOAuth({
```

Comment:
To eliminate possible encoding issues, this could be written something like:

```elixir
ui.initOAuth(<%=
  keys = [
    :client_id,
    :client_secret,
    :realm,
    :app_name,
    :scope_separator,
    :additional_query_string_params,
    :use_basic_authentication_with_access_code_grant,
    :use_pkce_with_authorization_code_grant,
  ]
  |> Enum.map(fn key ->
    camel_key = key |> to_string() |> String.replace(~r/_\w/, fn sep -> sep |> String.replace("_", "") |> String.upcase() end)

    if Keyword.has_key?(oauth, key) do
      {camel_key, oauth[key]}
    else
      nil
    end
  end)
  |> Enum.filter(& &1)
  |> Map.new()
  |> OpenApiSpex.OpenApi.json_encoder().encode()
  |> elem(1)
%>)
```

Reply:
Nice! 
I was hoping there was a pattern for turning an Elixir map / Keyword into JSON within an EEx template.